File: Unidad_1/Desbordamiento_silencioso.java
package desbordamiento_silencioso;
public class Desbordamiento_silencioso {
    public static void main(String[] args) {
        int max = Integer.MAX_VALUE; // 2,147,483,647
        int resultado = max + 10000;
        System.out.println("Maximo int: " + max);
        System.out.println("Maximo + 10: " + resultado);
        try {
            Math.addExact(max, 10);
        } catch (ArithmeticException e) {
            System.out.println("Error detectado: Desbordamiento Erp");
        }

    }
}

File: Unidad_1/3-Codigos/Desbordamiento_silencioso.py
def main():
    max_int = 2147483647 # Equivalente a Integer.MAX_VALUE en Java
    resultado = max_int + 10000
    
    print(f"Maximo int: {max_int}")
    print(f"Maximo + 10000: {resultado}") # En Python, esto suma perfectamente sin desbordarse.
    
    # Simulando el desbordamiento que capturaría Math.addExact
    try:
        suma_exacta = max_int + 10
        if suma_exacta > 2147483647:
            raise OverflowError("Error detectado: Desbordamiento Erp")
    except OverflowError as e:
        print(e)

if __name__ == "__main__":
    main()